apis/r/R/Factory.R

```diff
@@ -4,15 +4,22 @@
 #' @param uri URI for the TileDB object
 #' @param schema schema Arrow schema argument passed on to DataFrame$create()
 #' @param index_column_names Index column names passed on to DataFrame$create()
+#' @param mode Ingestion mode: one of \code{write} or \code{resume}
 #' @param platform_config Optional platform configuration
 #' @param tiledbsoma_ctx Optional SOMATileDBContext
 #' @param tiledb_timestamp Optional Datetime (POSIXct) for TileDB timestamp
 #' @export
 SOMADataFrameCreate <- function(uri, schema, index_column_names = c("soma_joinid"),
-                                platform_config = NULL, tiledbsoma_ctx = NULL, tiledb_timestamp = NULL) {
-    sdf <- SOMADataFrame$new(uri, platform_config, tiledbsoma_ctx, tiledb_timestamp, internal_use_only = "allowed_use")
-    sdf$create(schema, index_column_names=index_column_names, platform_config=platform_config, internal_use_only = "allowed_use")
-
+                                mode = "write", platform_config = NULL,
+                                tiledbsoma_ctx = NULL, tiledb_timestamp = NULL) {
+    spdl::debug("[SOMADataFrameCreate] mode={}", mode)
+    sdf <- SOMADataFrame$new(uri, platform_config, tiledbsoma_ctx, tiledb_timestamp,
+                             internal_use_only = "allowed_use")
+    if (!(mode == "resume" &&
+          tiledb::tiledb_vfs_is_dir(uri = sdf$uri, vfs = sdf$tiledbsoma_ctx$vfs()))) {
```

Comment:
@eddelbuettel I don't believe this works with `tiledb://...` URIs -- various SOMA types have `exists()` methods though ...

I'll play around a bit with `tiledb://...` URIs -- and I'll put in some examples you can use as well -- we do unit-test `tiledb://...` URIs in-package since this is an open-source package, so some extra manual validation is valuable here.

Reply:
Yes it is complicate.   But note that I did not change any VFS logic, it is all exactly as before.  

What I did change now was to instantiate a VFS object based off the currently-used Ctx object.  That should allow for example S3 regions set to transfer accordingly to this access etc.   

So I don't think this puts up new barriers.   We just check URIs that we know we can write to (and would in the next line) for whether they already exist -- which was the task of the PR.

But yes: if this needs one of those 'tiledb:// -->  s3://' translations we would have to do it here.